lib/avrora/storage/registry.ex

```diff
@@ -93,20 +93,67 @@ defmodule Avrora.Storage.Registry do
   @spec configured?() :: true | false
   def configured?, do: !is_nil(Config.registry_url())
 
+  defp handle_http_client(path, http_client_function) do
+    with {:ok, url} <- handle_url(path),
+         {:ok, headers} <- handle_headers() do
+      url |> http_client_function.(headers) |> handle()
+    end
+  end
+
   defp http_client_get(path) do
+    handle_http_client(path, fn url, headers ->
+      http_client().get(url, headers: headers)
+    end)
+  end
+
+  defp http_client_post(path, payload) do
+    handle_http_client(path, fn url, headers ->
+      http_client().post(url, payload, headers: headers, content_type: @content_type)
+    end)
+  end
+
+  defp handle_url(path) do
     if configured?(),
-      do: path |> to_url() |> http_client().get() |> handle(),
+      do: {:ok, to_url(path)},
       else: {:error, :unconfigured_registry_url}
   end
 
-  defp http_client_post(path, payload) do
-    if configured?() do
-      path |> to_url() |> http_client().post(payload, content_type: @content_type) |> handle()
-    else
-      {:error, :unconfigured_registry_url}
+  defp handle_headers do
+    with {:ok, authorization} <- auth() do
+      {:ok, [authorization]}
+    end
+  end
+
+  defp auth do
+    case registry_auth() do
+      nil ->
+        {:ok, []}
+
+      {:basic, file_path} = auth when is_binary(file_path) ->
```

Comment:
I think this should be just 1-time operation in `Config` module, but not every call to the registry

Reply:
I think that the code will be more organized, but even if we move the encoding function to the `Config` module, this function will still be called on every call to the registry. Maybe this will add more complexity, but what if we cache the authentication with `:ets` to be a 1-time operation during n time?

Reply:
Yes, you are right, but reading a file and reading the ENV variable (or so) is very different on performance. And, in case if you need to read N schemas it will be N file reads.

Reply:
Oh, you were referring to the file read function. Since we'll scope out it then I think this will not be a problem anymore. But still, I'll move the encoding function to the `Config` module:

```elixir
  @doc false
  def registry_basic_auth do
    with [user, pass] <- get_env(:registry_basic_auth, nil) do
      base64 = :base64.encode_to_string("#{user}:#{pass}")
      {'Authorization', 'Basic #{base64}'}
    end
  end
```